packages/poseidon-proof/tests/index.test.ts

```diff
@@ -1,85 +1,40 @@
 import { buildBn128 } from "@zk-kit/groth16"
 import { decodeBytes32String, toBeHex } from "ethers"
-import {
-    poseidon1,
-    poseidon10,
-    poseidon11,
-    poseidon12,
-    poseidon13,
-    poseidon14,
-    poseidon15,
-    poseidon16,
-    poseidon2,
-    poseidon3,
-    poseidon4,
-    poseidon5,
-    poseidon6,
-    poseidon7,
-    poseidon8,
-    poseidon9
-} from "poseidon-lite"
+import { poseidon2 } from "poseidon-lite"
 import generate from "../src/generate"
 import hash from "../src/hash"
 import { PoseidonProof } from "../src/types"
 import verify from "../src/verify"
 
-const poseidonFunctions = [
-    poseidon1,
-    poseidon2,
-    poseidon3,
-    poseidon4,
-    poseidon5,
-    poseidon6,
-    poseidon7,
-    poseidon8,
-    poseidon9,
-    poseidon10,
-    poseidon11,
-    poseidon12,
-    poseidon13,
-    poseidon14,
-    poseidon15,
-    poseidon16
-]
-
-const computePoseidon = (preimages: string[]) => poseidonFunctions[preimages.length - 1](preimages)
-
-const preimages = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]
 const scope = "scope"
 let curve: any
-const proofs: Array<{ fullProof: PoseidonProof; digest: bigint }> = []
+let digest: bigint
+let fullProof: PoseidonProof
 
 beforeAll(async () => {
     curve = await buildBn128()
-    for (const preimage of preimages) {
-        const currentPreimages = preimages.slice(0, preimage)
-        const fullProof = await generate(currentPreimages, scope)
-        const digest = computePoseidon(currentPreimages.map((preimage) => hash(preimage)))
-        proofs.push({ fullProof, digest })
-    }
-}, 180_000)
+
+    fullProof = await generate([1, 2], scope)
```

Comment:
this will definitely be a faster test!
I know we are having lot of timeout issues atm.
I am just worried about reducing test coverage (each generate function uses different artifacts files) just to satisfy the ci...

Reply:
I've been thinking about this. I think we should ask ourselves what we are actually testing in this library.

If the artifacts are based on the circuit in `@zk-kit/circuits` (whose parameters should be tested there), they should be fine, and we don't need to test them here (a function to check the integrity should be enough). Given that, 1 couple of artifacts should be enough to test the library logic I guess.

Does it make sense? I may be missing something tho.

Reply:
> @zk-kit/circuits (whose parameters should be tested there)
> we don't need to test them here
> Given that, 1 couple of artifacts should be enough to test the library logic I guess.

I agree. Testing a circuit works for different parameters is in the scope of the circuit package.

> a function to check the integrity should be enough

And we said this integrity check is in the scope of https://github.com/privacy-scaling-explorations/snark-artifacts right? see https://github.com/privacy-scaling-explorations/snark-artifacts/issues/11

Reply:
> And we said this integrity check is in the scope of https://github.com/privacy-scaling-explorations/snark-artifacts right? see https://github.com/privacy-scaling-explorations/snark-artifacts/issues/11

Yes!